### ISSUE TITLE
Simpler Reorderable Lists Api

### ISSUE THREAD

**Issue by VisenDev** [david-vanderson/dvui#109]
The `simple` reorderable list example still seems quite complex. I struggled to understand the simple and advanced usages in `Example.zig`

For example, it seems like this function could be automatically generated with comptime, and shouldn't need to be manually written by the user
```zig
        pub fn reorder(removed_idx: ?usize, insert_before_idx: ?usize) void {
            if (removed_idx) |ri| {
                if (insert_before_idx) |ibi| {
                    // save this index
                    const removed = strings[ri];
                    if (ri < ibi) {
                        // moving down, shift others up
                        for (ri..ibi - 1) |i| {
                            strings[i] = strings[i + 1];
                        }
                        strings[ibi - 1] = removed;
                    } else {
                        // moving up, shift others down
                        for (ibi..ri, 0..) |_, i| {
                            strings[ri - i] = strings[ri - i - 1];
                        }
                        strings[ibi] = removed;
                    }
                }
            }
        }
```

It would be really nice to have some reorderable list helper functions which would make this widget much more convenient to use.

For example, something along the lines of
```zig
pub fn reorderList(src: std.builtin.SourceLocation, comptime T: type, list_items: []T,  list_item_widget: *fn(item: *T, index: usize) !void) !ReorderableList(T){}
```
might be possible

**Comment by david-vanderson** [comment]
Yes you are totally right.  I wrote the Reorderable stuff to be as generic as possible, which is useful.  But a simple function for the normal case given a slice is a great idea.  Will do.

**Comment by david-vanderson** [comment]
I added a helper function `reorderSlice` like you suggested, and that looks good.

I spent a lot of time trying to figure out a useful function that did it all in one go, but as of now I'm not sure it makes sense.  Like you wrote, it would have to take a callback to render the entry in the list but that seems as complicated as directly using the widget.  I'm also struggling to think of a usecase where the containing code didn't care when things got reordered.

But this did give me the nudge to make ReorderWidget a normal widget (it had been a kind of half-way widget from during development).

I think we'll have to wait for some usecases to show up to understand if a wrapping function is worthwhile.  Sound good?

**Comment by VisenDev** [comment]
> I think we'll have to wait for some usecases to show up to understand if a wrapping function is worthwhile. Sound good?

Thanks! I will experiment with the new widget in my structEntry pr and see how it ends up working.
